### PR TITLE
feat: reimplement directory sidebar rendering to reduce calls to DB

### PR DESCRIPTION
**Pull request by karrui**
## Solution

This PR refactors the CMS sidebar navigation component to improve performance and maintainability.

**Breaking Changes**

- [x] No - this PR is backwards compatible

**Features**:

- Introduced a new `DirectorySidebar` component to replace the `CmsSideNav`
- Implemented lazy loading of child resources to reduce initial load time
- Added a "Home" sublabel for the root page entry

**Improvements**:

- Separated concerns by splitting the sidebar logic into multiple components
- Improved code readability and maintainability
- Enhanced performance by only fetching children when necessary

**Bug Fixes**:

- Fixed an issue where the sidebar would fetch for all first-level children on load

## Before & After Screenshots
See Storybook

### CODE REVIEW

**Comment by karrui** [comment]
* **#729** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/729?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/studio/tests/msw/handlers/resource.ts [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
+            | "Folder"
+            | "Collection"
+            | "CollectionPage",
+          // ID must be unique so infinite loop won't occur
+          id: `${resourceId}-${item.title}-${item.id}`,
+        }))
+
+        // if (resourceId !== null) {
+        //   items =
+        // }
```
Comment (seaerchin) [inline apps/studio/tests/msw/handlers/resource.ts:22]: wutface

**Review by seaerchin — APPROVED** [review]
lgmt but i never try yet

File: apps/studio/tests/msw/handlers/resource.ts (hunk 1 above)
Comment (karrui) [reply apps/studio/tests/msw/handlers/resource.ts:22]: oops i will remove

**Pull request merged**